static-assets/components/cstudio-forms/data-sources/file-desktop-upload.js

```diff
@@ -101,48 +101,18 @@ YAHOO.extend(CStudioForms.Datasources.FileDesktopUpload, CStudioForms.CStudioFor
   },
 
   edit: function (key, control) {
-    this._self = this;
-    var me = this;
-
-    var site = CStudioAuthoringContext.site;
-    var path = this._self.repoPath;
-    var isUploadOverwrite = true;
-
-    for (var i = 0; i < this.properties.length; i++) {
-      if (this.properties[i].name === 'repoPath') {
-        path = this.properties[i].value;
-
-        path = this.processPathsForMacros(path);
```

Comment:
What about all this removed code, including parsing the path macros, etc?

Reply:
The file upload behavior when editing was to upload/override the item (removed code). From the ticket, the behavior should be displaying the editor (if editable). That's what the browse-file datasource does.